clevertap-core/src/main/java/com/clevertap/android/sdk/LocalDataStore.java

```diff
@@ -53,6 +57,7 @@ public class LocalDataStore {
 
     private final DeviceInfo deviceInfo;
     private final Set<String> userEventLogKeys = Collections.synchronizedSet(new HashSet<>());
+    private final Map<String, String> normalizedEventNames = new HashMap<>();
```

Comment:
Do we have this to reduce repeated calls to `getNormalizedName`?
Seems like a tradeoff between memory and execution-time

Reply:
>Do we have this to reduce repeated calls to getNormalizedName?

YES. To improve performance. Memoize already computed stuffs.

Reply:
Memory will not be an issue here. Memory will be released once App is terminated.

Reply:
Common events can occur multiple times in the same session like `Product Viewed`, `Add to cart`, `Product Details Viewed`, `Item increment`, `Item decrement` and many. Also it's not just events, we have huge payload of CS in-apps in which case same event name can appear multiple times. CS in-apps can have 100 in-apps, each in-app can have multiple triggers, in that way I believe it's better to memoize. Also If we put `log()` in `Utils.getNormalizedName()` we can see it get's call multiple times for already computed value.